src/Pinch/Internal/RPC.hs

```diff
@@ -40,3 +48,7 @@ readMessage chan = Transport.readMessage (cTransportIn chan) $ deserializeMessag
 
 writeMessage :: Channel -> Message -> IO ()
 writeMessage chan msg = Transport.writeMessage (cTransportOut chan) $ serializeMessage (cProtocolOut chan) msg
+
+class (Pinchable a, Tag a ~ TStruct) => ThriftResult a where
+  type ResultType a
+  toEither :: a -> Either SomeException (ResultType a)
```

Comment:
Example code;

```
data Success_Result
  = Success_Result_Success Data.Text.Text
  deriving (Prelude.Eq, GHC.Generics.Generic, Prelude.Show)

instance Pinch.Pinchable Success_Result where
  .....

instance Pinch.ThriftResult Success_Result where
  type (ResultType Success_Result) = Data.Text.Text
  toEither (Success_Result_Success x) = Prelude.Right (x)
```

Reply:
Hm, on second thoughts, this could also be encoded more directly:

```
class .... ThriftResult a where
  extract :: a -> IO (ResultType a)

instance Pinch.ThriftResult Example_Result where
  type (ResultType Example_Result) = Data.Text.Text
  extract (Example_Result_Success x) = pure x
  extract (Example_Result_MyError e) = throwIO e
```

What do you think?

Reply:
I like the exception version since that's how we're surfacing failures in the
rest of the client/server logic.

FWIW, that approximately matches the helper we generate for this purpose in
our Go variant: a `UnwrapResponse` function that accepts the `*_Result` type
and returns either the return value of that function, or the exception as an
error.